ui/js/components/auth.js

```diff
@@ -117,70 +121,24 @@ treeherder.component("login", {
             };
 
             ctrl.setLoggedIn = function (newUser) {
+                const userSession = localStorageService.get('userSession');
+
                 newUser.loggedin = true;
-                localStorageService.set("user", newUser);
+                newUser.fullName = userSession.fullName;
+
                 ctrl.user = newUser;
                 ctrl.onUserChange({ $event: { user: newUser } });
 
+                // start session renewal process
+                if (userSession && userSession.renewAfter) {
+                    authService.resetRenewalTimer(userSession);
+                }
             };
 
             ctrl.setLoggedOut = function () {
-                localStorageService.set("user", loggedOutUser);
-                localStorageService.set('taskcluster.credentials', {});
+                authService.logout();
                 ctrl.user = loggedOutUser;
                 ctrl.onUserChange({ $event: { user: loggedOutUser } });
             };
         }]
 });
-
-treeherder.component("loginCallback", {
-    template: `
-        <div ng-if="!loginError">
-            <span>Logging in... <span class="fa fa-spinner fa-spin"></span></span>
-        </div>
-        <div ng-if="loginError">
-            <span>Error logging in: {{loginError}}</span>
-        </div>
-    `,
-    controller: ['localStorageService', '$location', '$window', '$http', '$scope',
-        function (localStorageService, $location, $window, $http, $scope) {
-            const hawk = require('hawk');
```

Comment:
This was the only direct usage of the `hawk` library, so it can now be removed from `package.json` and `neutrino-custom/base.js`

Reply:
Removing the `hawk` package with `yarn remove hawk` is causing the browser to throw an error when navigating to `http://localhost:8000`. I get:

```
TypeError: a.a.crypto.utils is undefined
```

Reply:
So looking into this it appears to be due to a combination of yarn not hoisting hawk to the `node_modules` root (there are multiple versions of hawk in use - it must not re-assess which should be hoisted on removal) + a Neutrino bug which will go away once we update to Neutrino 8.

As such let's leave these alone for now. Sorry for the churn!